cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
-`style-src 'unsafe-inline' https:;`
+# CSP Sample Policies
 
-This is what was used at Twitter, Oct 2014. The policy prevents mixed content, allows for scheme "data:" in font-src and img-src, allows for unsafe-inline and unsafe-eval for script-src, and unsafe-inline for style-src. (see: <https://twittercommunity.com/t/blocking-mixed-content-with-content-security-policy/26375>)
+## Basic CSP Policy
 
-Mixed Content has two categories: Active and Passive. Passive content consists of "resources which cannot directly interact with or modify other resources on a page: images, fonts, audio, and video for example", whereas active content is "content which can in some way directly manipulate the resource with which a user is interacting." (http://www.w3.org/TR/2014/WD-mixed-content-20140722)
+This policy will only allow resources from the originating domain for all the default level directives, and will not allow inline scripts/styles to execute. If your application and function with these restrictions, it drastically reduces your attack surface having this policy in place, and will work with most modern browsers.
 
-`Content-Security-Policy: img-src https: data:; font-src https: data:; media-src https:;`
+The most basic policy assumes:
 
-This is an example to block only passive mixed content.
+-   All resources are hosted by the same domain of the document.
+-   There are no inlines or evals for scripts and style resources.
 
-`Content-Security-Policy: script-src https:; style-src https:; object-src https:; connect-src https:; frame-src https:; `
+> `Content-Security-Policy: default-src 'self';`
 
-This is an example to block only active mixed content.
+To tighten further, one can do the following:
 
-Preventing Clickjacking
------------------------
+> `Content-Security-Policy: default-src 'none'; script-src 'self'; connect-src 'self'; img-src 'self'; style-src 'self';`
 
-The established way of preventing clickjacking involves the use of the header `X-Frame-Options` (see: [Clickjacking_Defense_Cheat_Sheet](/Clickjacking_Defense_Cheat_Sheet "wikilink")). However, CSP 2.0 has a new directive `frame-ancestors`.
+This policy allows images, scripts, AJAX, and CSS from the same origin, and does not allow any other resources to load (eg. object, frame, media, etc).
 
-To prevent all framing of your content use:
+## Mixed Content Policy
 
-`Content-Security-Policy: frame-ancestors 'none'`
+- In order to prevent mixed content (resources being loaded over http, from a document loaded over https), one can use the following directive to block mixed content.
 
-To allow for your site only, use:
+  - `Content-Security-Policy: block-all-mixed-content;`
 
-`Content-Security-Policy: frame-ancestors 'self'`
+- On the other hand, if the developer is migrating from HTTP to HTTPS, the following directive will ensure that all requests will be sent over HTTPS with no fallback to HTTP:
```

Comment:
add ref to the [directive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/upgrade-insecure-requests) doc.